### PR TITLE
Add prose to section headers to guide the reader

### LINKED ISSUE #334: Add more prose
Many of the sections in the document simply start with function or data structure descriptions without much context. We should add more prose so that the document is easier to follow and understand.

### PR DESCRIPTION
**Pull request by chris-wood**
Closes #334.

### CODE REVIEW

**Review by bytemare — APPROVED** [review]

File: draft-irtf-cfrg-opaque.md [hunk 1]
```diff
@@ -731,13 +732,24 @@ The registration protocol then runs as shown below:
...
+
 Both client and server MUST validate the other party's public key before use.
 See {{validation}} for more details. Upon completion, the server stores
 the client's credentials for later use. Moreover, the client MAY use the output
 `export_key` for further application-specific purposes; see {{export-key-usage}}.
 
 ## Registration Messages {#registration-messages}
 
+This section contains definitions of the RegistrationRequest,
+RegistrationResponse, and RegistrationRecord messages exchanged between
```
Comment (kevinlewi) [inline draft-irtf-cfrg-opaque.md:750]: These should be displayed using backticks right?

File: draft-irtf-cfrg-opaque.md [hunk 2]
```diff
@@ -777,8 +789,14 @@ envelope: The client's `Envelope` structure.
 
 ## Registration Functions {#registration-functions}
 
+This section contains definitions of the functions used by client and server
+during registration, including CreateRegistrationRequest, CreateRegistrationResponse,
+and FinalizeRegistrationRequest.
```
Comment (kevinlewi) [inline draft-irtf-cfrg-opaque.md:794]: Same here regarding backticks, though I see we are less consistent about function names and backticking them. Can you also backtick any occurrences of function names?

File: draft-irtf-cfrg-opaque.md [hunk 3]
```diff
@@ -1039,6 +1049,9 @@ in later sections:
 
 ### ClientInit
 
+The ClientInit function begins the AKE protocol and produces the client's KE1
```
Comment (kevinlewi) [inline draft-irtf-cfrg-opaque.md:1052]: backticks for ClientInit and KE1

File: draft-irtf-cfrg-opaque.md [hunk 4]
```diff
@@ -1060,6 +1073,9 @@ def ClientInit(password):
 
 ### ServerInit
 
+The ServerInit function continues the AKE protocol by processing the client's KE1 message
+and producing the server's KE2 output.
```
Comment (kevinlewi) [inline draft-irtf-cfrg-opaque.md:1077]: ServerInit, KE1, KE2 backticks

File: draft-irtf-cfrg-opaque.md [hunk 5]
```diff
@@ -1088,6 +1104,10 @@ def ServerInit(server_identity, server_private_key, server_public_key,
 
 ### ClientFinish
 
+The ClientFinish function completes the AKE protocol for the client and
+produces the client's KE3 output for the server, as well as the `session_key`
```
Comment (kevinlewi) [inline draft-irtf-cfrg-opaque.md:1108]: ClientFinish, KE3 backticks

File: draft-irtf-cfrg-opaque.md [hunk 6]
```diff
@@ -1118,11 +1138,12 @@ def ClientFinish(client_identity, server_identity, ke2):
 
 ### ServerFinish
 
-Since the OPRF is a two-message protocol, KE3 has no element of the OPRF. We can
-therefore call the AKE's `AuthServerFinalize` directly. The `AuthServerFinalize` function
-MUST take `KE3` as input and MUST verify the client authentication material it contains
-before the `session_key` value can be used. This verification is paramount in order to
-ensure forward secrecy against active attackers.
+The ServerFinish function completes the AKE protocol for the server, yielding the `session_key`.
```
Comment (kevinlewi) [inline draft-irtf-cfrg-opaque.md:1141]: ServerFinish backticks

File: draft-irtf-cfrg-opaque.md [hunk 7]
```diff
@@ -1118,11 +1138,12 @@ def ClientFinish(client_identity, server_identity, ke2):
...
+Since the OPRF is a two-message protocol, KE3 has no element of the OPRF, and it
```
Comment (kevinlewi) [inline draft-irtf-cfrg-opaque.md:1142]: KE3 backticks

File: draft-irtf-cfrg-opaque.md [hunk 8]
```diff
@@ -1146,8 +1167,14 @@ ClientAuthenticationError, invoked from `AuthServerFinalize`.
 
 ## Credential Retrieval {#cred-retrieval}
 
+This section describes the sub-protocol run during authentication to retrieve and
+recover the client credentials.
+
 ### Credential Retrieval Messages
 
+This section describes the CredentialRequest and CredentialResponse messages exchanged
```
Comment (kevinlewi) [inline draft-irtf-cfrg-opaque.md:1175]: CredentialRequest and CredentialResponse backticks

File: draft-irtf-cfrg-opaque.md [hunk 9]
```diff
@@ -1174,8 +1201,14 @@ client's `Envelope` structure.
 
 ### Credential Retrieval Functions
 
+This section describes the CreateCredentialRequest, CreateCredentialResponse,
+and RecoverCredentials functions used for credential retrieval.
```
Comment (kevinlewi) [inline draft-irtf-cfrg-opaque.md:1205]: Backticks for function names

File: draft-irtf-cfrg-opaque.md [hunk 10]
```diff
@@ -1174,8 +1201,14 @@ client's `Envelope` structure.
...
+
 #### CreateCredentialRequest {#create-credential-request}
 
+The CreateCredentialRequest is used by the client to initiate the credential
+retrieval process, and it produces a CredentialRequest message and OPRF state.
```
Comment (kevinlewi) [inline draft-irtf-cfrg-opaque.md:1210]: Backticks for function names and message names

**Review by kevinlewi — APPROVED** [review]
We are doing backticks for all function and message names, right? Just checking... I did not mark all of the ones towards the end of the file, but if this is the case, please update!

Otherwise, looks good :)

**Comment by chris-wood** [comment]
> We are doing backticks for all function and message names, right? Just checking... I did not mark all of the ones towards the end of the file, but if this is the case, please update!

Backticked everything! Will resolve conflicts and then merge.

**Pull request merged**